src/Compilers/CSharp/Portable/Operations/CSharpConversionExpression.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Semantics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal abstract class BaseCSharpConversionExpression : BaseConversionExpression
+    {
+        protected BaseCSharpConversionExpression(Conversion conversion, bool isExplicitInCode, bool throwsExceptionOnFailure, bool isChecked, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue) :
+            base(conversion, isExplicitInCode, throwsExceptionOnFailure, isChecked, syntax, type, constantValue)
+        {
+            ConversionInternal = conversion;
+        }
+
+        internal Conversion ConversionInternal { get; }
+
+        public override string LanguageName => LanguageNames.CSharp;
```

Comment:
I'm of the opinion that we need it somewhere. Otherwise, the expected path for me to use the API is to call one of our `GetConversion` methods, catch the resulting `ArgumentException` if the source code wasn't that one, and then try the other one. Our users need _something_ to switch off of.